Source/Dafny/AST/DafnyAst.cs

```diff
@@ -5953,29 +5899,17 @@ public abstract class NonglobalVariable : IVariable {
         return name;
       }
     }
-    public string DisplayName {
-      get { return LocalVariable.DisplayNameHelper(this); }
-    }
+    public string DisplayName =>
+      LocalVariable.DisplayNameHelper(this);
+
     private string uniqueName;
-    public string UniqueName {
-      get {
-        return uniqueName;
-      }
-    }
-    public bool HasBeenAssignedUniqueName {
-      get {
-        return uniqueName != null;
-      }
-    }
+    public string UniqueName => uniqueName;
+    public bool HasBeenAssignedUniqueName => uniqueName != null;
     public string AssignUniqueName(FreshIdGenerator generator) {
-      if (uniqueName == null) {
-        uniqueName = generator.FreshId(Name + "#");
-        compileName = string.Format("_{0}_{1}", IVariable.CompileNameIdGenerator.FreshNumericId(), CompilerizeName(name));
-      }
-      return UniqueName;
+      return uniqueName ??= generator.FreshId(Name + "#");
     }
     static char[] specialChars = new char[] { '\'', '_', '?', '\\', '#' };
-    public static string CompilerizeName(string nm) {
+    public static string SanitizeName(string nm) {
```

Comment:
Could you add a comment to this method? What does it mean to sanitize a name? Why do we need to sanitize? Apparently the verifier also needs sanitized names. Does SanitizeName remove any characters which are likely to cause issues in verification or compilation back-ends ?

Reply:
IIUC what this does is indeed escape some special chars that shouldn't be emitted into Boogie. (The notion of "sanitized" names already existed, so I just renamed the function for consistency.)

As for compilation versus verification, I'm less clear on that.  We already have idProtect on the compiler side, so I'm guessing SanitizeName isn't strictly needed there.

Reply:
6e69a31ea0703e3dc35b103a9f85cf0286f4d49b was where it was originally introduced, and that was specifically for C#.  Then the translator was changed to use "SanitizedName" (an alias of CompileName, more or less) in a8b18b984b289be43c112ca6f0fb2283a8ddc13a. IdProtect was introduced in a939eff5 to sanitize language-specific keywords.

Reply:
I added a docstring.